configure.ac

```diff
@@ -168,6 +174,7 @@ AC_CONFIG_FILES([
 	module/unicode/Makefile
 	module/zcommon/Makefile
 	module/zfs/Makefile
+	module/zstd/Makefile
```

Comment:
Please, let's not call this `zstd` in the kernel space. Can we prefix this to avoid name collisions?

Reply:
Testing has proven no issues with so ever doing this.
This is afaik also required for operations.
@c0d3z3r0 or @brainslayer might be beter suited to explain.

Reply:
Yeah, no. This is not okay. We already introduce enough potential problems by doing dumb things like naming modules like `icp` and `unicode`. Let's _not_ make this worse with `zstd`. So many things can go wrong here...

Reply:
@Conan-Kudo We seem to have some misunderstanding here. You are talking about the module name, not about calling (running) zstd in kernel-space, right?

Reply:
Yes, the module name and (hopefully) the symbols exported by the module...

Reply:
As @brainslayer suggested it might be prudent (albeit not required) renaming the ZFS side of the ZSTD implementation, which solve most issues related to this.

Reply:
However:
@Conan-Kudo The module name (as defined in the makefile) is actually zzstd, so there shouldn't be a conflict with zstd.

We could rename all of the zfs side zstd code to zzstd, but I think it's clear:
If it ain't in contrib, its zfs-side, if it is in contrib it's native zstd.

Reply:
note: the zzstd name was choosen in early development by request from zfs maintainers

Reply:
True, thats the module name not the folder structure...

If we change anything it's renaming ALL:
Files, Folders and references.

Reply:
Anyhow: If the maintainers also want all file and folder names changed to zzstd, i'm fine...
But lets keep the choice with them.

Reply:
ack